### ISSUE TITLE
Feature Request: async preprocess

### ISSUE THREAD

**Issue by FlorianWendelborn** [colinhacks/zod#1014]
I’m trying to parse some minecraft NBT data with zod. However, in order to do this, it first needs to be converted from the binary NBT format to a JSON-like structure.

The best way to implement this in my opinion would be to use zod’s `z.preprocess()`. Unfortunately, the library I’m using [`prismarine-nbt`](https://www.npmjs.com/package/prismarine-nbt) only exposes `parse` as an `async` function.

As far as I can tell, async preprocessing is not supported right now. I think it would be great if it’s allowed as it would really simplify parsing with `async` libraries.

**Comment by colinhacks** [comment]
Yep, seems like an oversight, should be a quick addition. I'll look into it.

**Comment by FlorianWendelborn** [comment]
@stale this issue is still relevant

**Comment by FlorianWendelborn** [comment]
@stale don’t

**Comment by FlorianWendelborn** [comment]
@stale don't

**Comment by FlorianWendelborn** [comment]
@stale don't

**Comment by FlorianWendelborn** [comment]
@stale don’t

**Comment by lennerd** [comment]
A simply workaround:

```ts
z.preprocess(
  async (value) => {
    // ...
  },
  z.promise(someSchema), //  Wrap your own schema with a promise schema.
);
```

**Comment by RobertGemmaJr** [comment]
Any updates on this?
